framework/doc/content/finite_volumes/linear_fv_design.md

```diff
@@ -0,0 +1,162 @@
+- `computeBoundaryRHSContribution` which computes the right hand side contributions of
+  a boundary face to the degree of freedom corresponding to the adjacent element.
+
+### Elemental Kernels:
+
+For volumetric effects, an element-based evaluation is necessary. Good examples are
+external source terms or reaction terms in the partial differential equations.
+To manage the contributions of these terms to the linear system, a
+`LinearFVElementalKernel` has been added which can be used for inheritance.
+The derived kernels need to override the following functions:
+
+- `computeMatrixContribution` which computes the matrix contributions to the degree of
+  freedom corresponding to the element.
+- `computeRightHandSideContribution` which computes the right hand side contributions
+  to the degree of freedom corresponding to the element.
+
+## Looping over Faces and Elements
+
+The population of the system matrix and right hand side with face terms
+(from flux kernels) is done in a face loop: `ComputeLinearFVFaceThread`.
+
+On the other hand, the population of the system matrix and right hand side
+with element-based volumetric terms is done in an element
+loop: `ComputeLinearFVElementalThread`.
+
+## Gradient Computation
+
+Considering that the linear finite volume system is used in a Picard-style iteration,
+terms that need cell gradients are lagged. Such terms include:
+
+- The linear terms in the extrapolated boundary conditions.
+- The nonorthogonal correctors in the surface normal gradient computation.
+
+The explicit treatment of these terms allows us to have one ghosted layer in
+parallel simulations, minimizing the linear solve time and parallel communication
+time over an iteration. The setbacks of this design choice are:
+
+- We need corrector iterations to resolve extrapolated boundary conditions
+  and nonorthogonal correctors. This means that even linear systems may need to be solved
+  multiple times to get a converged solution.
+- It may increase the number of iterations needed to resolve nonlinearities.
+
+## Boundary Conditions
+
+A significant difference compared to other systems in MOOSE is the way
+boundary conditions are enforced. In the linear finite volume setting,
+boundary conditions are enforced through `LinearFVFluxKernel`, using the following
+two functions:
+
+- `computeBoundaryMatrixContribution` which computes the matrix contributions of
+  a boundary face to the degree of freedom corresponding to the adjacent element.
+- `computeBoundaryRHSContribution` which computes the right hand side contributions of
+  a boundary face to the degree of freedom corresponding to the adjacent element.
+
+For these types of boundary conditions, a new system has been created and objects can be
+created in the input file under the `[LinearFVBCs]` block. To create a new boundary
+condition the following functions need to be overridden:
+
+- `computeBoundaryValue` computes the boundary value of the field.
```

Comment:
I'm sure it will become clear once I'm looking at the input files. After finishing reading this page, I suspect that a combination of objects under the two blocks will pair to enforce the condition

Reply:
I clarified this in the first paragraph and truncated the second one:
```
A significant difference compared to other systems in MOOSE is the way
boundary conditions are enforced. Even though the user is still required to define the
boundary conditions in the `[LinearFVBCs]` block, these boundary conditions
are not executed on their own; instead they are used within `LinearFVFluxKernel`-s, through the following two functions:
```